addons/mail/static/src/components/attachment_viewer/attachment_viewer.js

```diff
@@ -136,7 +136,7 @@ class AttachmentViewer extends Component {
      * @private
      */
     _close() {
-        this.attachmentViewer.delete();
+        this.attachmentViewer.attachment.closeView();
```

Comment:
instead make `_close` public, modify so it calls `.close()` of dialog manager, and call it from here